org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaClientConnection.java

```diff
@@ -56,6 +61,10 @@ public JavaClientConnection(JavaLanguageClient client) {
 		logHandler.install(this);
 	}
 
+	public Object executeClientCommand(String id, Object... params) {
+		return this.client.executeClientCommand(new ExecuteCommandParams(id, ImmutableList.copyOf(params))).join();

```

Comment:
should probably add a timeout

Reply:
That raises the question... how long should the timeout be? Or should we change this api and return the CompletableFuture? Then the caller can decide how long they want to wait... or if they want to continue with an async composition instead.

Reply:
Hmmm... returning CompletableFuture seems a bit 'out of style' with the rest of the api in the same class. I called join because I as trying to follow similar pattern as other stuff in the same class. E.g. like here:

https://github.com/kdvolder/eclipse.jdt.ls/blob/66aa25d44cea30d2fdf90b30058508772b990b48/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaClientConnection.java#L107